### PR TITLE
BOAC-4196: Allows cats, subcats to be linked to unit reqts; BOAC-4250

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-4196

Also fixes a bug with creating/updating unit requirements in the template builder:
https://jira.ets.berkeley.edu/jira/browse/BOAC-4250

### CODE REVIEW

File: src/components/degree/EditCategory.vue [hunk 1]
```diff
@@ -249,9 +250,18 @@ export default {
...
-      this.$announcer.polite(option ? `${this.selectedParentCategory} selected` : 'Unselected')
+      this.$announcer.polite(option ? `${this.selectedParentCategory.name} selected` : 'Unselected')
+      const existingUnitRequirements = this.selectedUnitRequirements
+      const parentUnitRequirements = this.$_.get(this.selectedParentCategory, 'unitRequirements')
+
       if (option) {
+        const inheritedUnitRequirements = this.$_.differenceBy(parentUnitRequirements, existingUnitRequirements, 'id')
+        this.$_.each(inheritedUnitRequirements, unitRequirement => {
+          this.$refs[`column-${option.position}-unit-requirement-select`].onChangeUnitRequirement(unitRequirement)
+        })
```
Comment (johncrossman) [inline src/components/degree/EditCategory.vue:261]: @lyttam - Is this the fix to BOAC-4250?  I.e., refresh what's in view when changes are made to unit-requirements? If so, I'd recommend using `$_refresh()` in `degree-edit-session.ts` instead.  We could add that refresh call in the `updateUnitRequirement` (vuex action).

Comment (lyttam) [reply src/components/degree/EditCategory.vue:261]: That's right @johncrossman . Good tip; I will amend.

Comment (lyttam) [reply src/components/degree/EditCategory.vue:261]: Correction: the change to UnitRequirements.vue line 245 is the fix for BOAC-4250. But I think your suggestion still makes sense.

Comment (johncrossman) [reply src/components/degree/EditCategory.vue:261]: @lyttam - It's unclear what line 260 is doing. Can you give short explanation?

Comment (lyttam) [reply src/components/degree/EditCategory.vue:261]: @johncrossman When the requirement location (parent category) is selected, that line triggers the change event on the SelectUnitFulfillment component for each unit reqt belonging to the parent. The effect is the inherited unit reqt(s) appear as though the user had selected them.

**Comment by johncrossman** [comment]
Thanks @lyttam

**Pull request merged**